src/coreclr/src/jit/hwintrinsic.h

```diff
@@ -315,6 +335,102 @@ struct HWIntrinsicInfo
     }
 };
 
+#ifdef TARGET_ARM64
+
+struct HWIntrinsic final
```

Comment:
The wrapper is used when we want to access the operands of GenTreeHWIntrinsic node. 

Otherwise, the code that does lookup:

```c++
        op1 = node->gtGetOp1();
        op2 = node->gtGetOp2();

        assert(op1 != nullptr);

        if (op1->OperIsList())
        {
            assert(op2 == nullptr);

            GenTreeArgList* list = op1->AsArgList();
            op1                  = list->Current();
            list                 = list->Rest();
            op2                  = list->Current();
            list                 = list->Rest();
            op3                  = list->Current();

            assert(list->Rest() == nullptr);

            numOperands = 3;
        }
        else if (op2 != nullptr)
        {
            numOperands = 2;
        }
        else
        {
            numOperands = 1;
        }
```
would need to be repeated in Lower, LSRA, CodeGen and, perhaps, other places.

I had this wrapper in CodeGen originally but here I decided to extend its use to the other places.

As an alternative, I can place this code directly in GenTreeHWIntrinsic (or even GenTreeJitIntrinsic).

Reply:
I think it might be cleaner to put it on one of the `GenTree` nodes.

Reply:
Okay, I can try this. Would you object me doing this as a separate PR and leave the wrapper as is here?

Reply:
No objection.